modules/nextflow/src/main/groovy/nextflow/splitter/AbstractTextSplitter.groovy

```diff
@@ -319,4 +327,21 @@ abstract class AbstractTextSplitter extends AbstractSplitter<Reader> {
      */
     abstract protected fetchRecord( BufferedReader reader )
 
+
+    protected int positionAfterBOM(Reader reader ){
+        if( !reader.markSupported() )
+            return 0
+        char[] beginner = new char[3];
+        reader.mark(beginner.length)
+        int read = reader.read(beginner)
+        reader.reset()
+        if( read ){
+            if( beginner[0] == '\ufeff' || beginner[0] == '\ufffe') {
```

Comment:
Out of curiosity, should not this handled by some groovy helper methods? If I'm not wrong I remember something like this implemented by methods in the  `DefaultGroovyMethods` extensions

Reply:
ummm no idea
I was looking if Groovy has something about it but all I've found redirects to Apache Commons so at the end I choose to implement it

Reply:
nvm